ironfish/src/mining/soloMiner.ts

```diff
@@ -171,15 +163,31 @@ export class MiningSoloMiner {
   private startNewWork(block: SerializedBlockTemplate) {
     Assert.isNotNull(this.currentHeadTimestamp)
     Assert.isNotNull(this.currentHeadDifficulty)
+    Assert.isNotNull(this.consensusParameters)
 
     const miningRequestId = this.nextMiningRequestId++
     this.miningRequestBlocks.set(miningRequestId, block)
     this.miningRequestId = miningRequestId
 
     this.target = Buffer.from(block.header.target, 'hex')
 
-    const work = mineableHeaderString(block.header)
-    this.newWork(miningRequestId, work)
+    const consensus = new Consensus(this.consensusParameters)
```

Comment:
Why is this being created every time a new work message comes in?

Reply:
Yea I can create it at initialization